metals/src/main/scala/scala/meta/internal/metals/MetalsServerConfig.scala

```diff
@@ -89,7 +90,12 @@ final case class MetalsServerConfig(
     ),
     bloopPort: Option[Int] = Option(System.getProperty("metals.bloop-port"))
       .filter(_.forall(Character.isDigit(_)))
-      .map(_.toInt)
+      .map(_.toInt),
+    macOsMaxWatchRoots: Int =
+      Option(System.getProperty("metals.macos-max-watch-roots"))
+        .filter(_.forall(Character.isDigit(_)))
+        .map(_.toInt)
+        .getOrElse(32)
```

Comment:
Could we default to 1? I find it more adequate assuming that most of the people work in small to medium-sized projects (pet projects, work ones) and only a few percent works with big monorepos. 
With 32 as a default, a lot of these ordinary people (not power-users) will see the message `INFO  Registering root for file watching: ...` being printed multiple times in Metals output and they may wonder why this suddenly changed. Sometimes it can be very misleading e.g. registering file watchers for sbt's aggregate projects will register watchers for phantom `src` and `test` in projects root (since it doesn't have sources or tests, it just aggregates other projects).

Reply:
For projects where the semantic db and class files are stored outside of the workspace root, this would result in a performance degradation when using the default configuration (their common parent would be watched). This can be solved by configuring Metals, but I feel like since 32 should be a safe value for everyone, this is an unnecessary step. But I'm perfectly open to changing it to 1 if you feel strongly about it.

Would you prefer the log message is moved to debug?

Reply:
Maybe let's change the message if `32` is safe. :+1:

Reply:
I set it to debug. Thank you!

Reply:
I was mostly worried about confusing messages, currently with the debug it's fine. Thanks!